dbms/src/Flash/Coprocessor/CHBlockChunkCodecV1.cpp

```diff
@@ -0,0 +1,202 @@
+
+
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <DataTypes/DataTypeFactory.h>
+#include <Flash/Coprocessor/CHBlockChunkCodecV1.h>
+
+#include <cassert>
+#include <cstddef>
+#include <vector>
+
+namespace DB
+{
+extern void WriteColumnData(const IDataType & type, const ColumnPtr & column, WriteBuffer & ostr, size_t offset, size_t limit);
+
+size_t ApproxBlockHeaderBytes(const Block & block)
+{
+    size_t size = 8 + 8; /// to hold some length of structures, such as column number, row number...
+    size_t columns = block.columns();
+    for (size_t i = 0; i < columns; ++i)
+    {
+        const ColumnWithTypeAndName & column = block.safeGetByPosition(i);
+        size += column.name.size();
+        size += 8;
+        size += column.type->getName().size();
+        size += 8;
+        if (column.column->isColumnConst())
+        {
+            size += column.column->byteSize() * column.column->size();
```

Comment:
why add byteSize * columnSize  as part of header here?

Reply:
removed.